crates/rbuilder/src/building/mod.rs

```diff
@@ -626,18 +622,10 @@ impl<Tracer: SimulationTracer> PartialBlock<Tracer> {
                 parse_deposits_from_receipts(&ctx.chain_spec, self.receipts.iter())
                     .map_err(|err| FinalizeError::Other(err.into()))?;
             let withdrawal_requests = system_caller
-                .post_block_withdrawal_requests_contract_call(
-                    db.as_mut(),
-                    &ctx.initialized_cfg,
-                    &ctx.block_env,
-                )
+                .post_block_withdrawal_requests_contract_call(db.as_mut(), &ctx.evm_env)
                 .map_err(|err| FinalizeError::Other(err.into()))?;
             let consolidation_requests = system_caller
-                .post_block_consolidation_requests_contract_call(
-                    db.as_mut(),
-                    &ctx.initialized_cfg,
-                    &ctx.block_env,
-                )
+                .post_block_consolidation_requests_contract_call(db.as_mut(), &ctx.evm_env)
```

Comment:
I'm pretty sure we need to call:

```
db.as_mut().merge_transitions(BundleRetention::Reverts);
```

somewhere in this block, e.g. before `Some(requests)` at line 641. h/t to Aresnii for spotting this on telegram!

Reply:
With the refactor this is now called down on line 654 (since the db is no longer shadowed).